stdlib/public/TensorFlow/Ops.swift

```diff
@@ -1597,52 +1597,257 @@ public extension Tensor where Scalar : Numeric {
+    return .range(Int32(self.lowerBound)..<Int32(self.upperBound), stride: 1)
+  }
+}
+
+extension ClosedRange : TensorSliceIndexProtocol where Bound == Int {
+  public var sliceIndex: TensorSliceIndex {
+    return .closedRange(Int32(self.lowerBound)...Int32(self.upperBound), stride: 1)
+  }
+}
+
+extension PartialRangeFrom : TensorSliceIndexProtocol where Bound == Int {
+  public var sliceIndex: TensorSliceIndex {
+    return .partialRangeFrom(Int32(self.lowerBound)..., stride: 1)
+  }
+}
+
+extension PartialRangeUpTo : TensorSliceIndexProtocol where Bound == Int {
+  public var sliceIndex: TensorSliceIndex {
+    return .partialRangeUpTo(..<Int32(self.upperBound), stride: 1)
+  }
+}
+
+extension PartialRangeThrough : TensorSliceIndexProtocol where Bound == Int {
+  public var sliceIndex: TensorSliceIndex {
+    return .partialRangeThrough(...Int32(self.upperBound), stride: 1)
+  }
+}
+
+public extension Tensor {
+  public struct IndexPath {
+    public let begin, end, strides: Tensor<Int32>
+    public let beginMask, endMask, ellipsisMask, newAxisMask, squeezeAxisMask: Int64
+    
+    @inlinable @inline(__always)
+    public init(
+      begin: Tensor<Int32>, end: Tensor<Int32>, strides: Tensor<Int32>, beginMask: Int64,
+      endMask: Int64, ellipsisMask: Int64, newAxisMask: Int64, squeezeAxisMask: Int64
+    ) {
+      self.begin = begin
+      self.end = end
+      self.strides = strides
+      self.beginMask = beginMask
+      self.endMask = endMask
+      self.ellipsisMask = ellipsisMask
+      self.newAxisMask = newAxisMask
+      self.squeezeAxisMask = squeezeAxisMask
+    }
+  }
+
+  @inlinable @inline(__always)
+  @differentiable(wrt: self, vjp: _vjpSubscript)
+  subscript(_ indexPath: IndexPath) -> Tensor {
+    get {
+      return Raw.stridedSlice(
+        self, begin: indexPath.begin, end: indexPath.end, strides: indexPath.strides,
+        beginMask: indexPath.beginMask, endMask: indexPath.endMask, 
+        ellipsisMask: indexPath.ellipsisMask, newAxisMask: indexPath.newAxisMask, 
+        shrinkAxisMask: indexPath.squeezeAxisMask)
+    }
+    set {
+      self = Raw.tensorStridedSliceUpdate(
+        self, begin: indexPath.begin, end: indexPath.end, strides: indexPath.strides,
+        value: newValue, beginMask: indexPath.beginMask, endMask: indexPath.endMask, 
+        ellipsisMask: indexPath.ellipsisMask, newAxisMask: indexPath.newAxisMask, 
+        shrinkAxisMask: indexPath.squeezeAxisMask)
+    }
+  }
+
+  @inlinable @inline(__always)
+  subscript(_ indices: TensorSliceIndexProtocol...) -> Tensor {
+    get {
+      return self[IndexPath(indices.map { $0.sliceIndex })]
+    }
+    set {
+      self[IndexPath(indices.map { $0.sliceIndex })] = newValue
+    }
+  }
+
+  @inlinable @inline(__always)
+  internal func _vjpSubscript(_ indexPath: IndexPath) -> (Tensor, (Tensor) -> Tensor) {
+    return (self[indexPath], { [shape = shapeTensor] v in
+      return Tensor<Scalar>._pullbackSubscript(v, indexPath, shape)
```

Comment:
Omit explicit return when it's the only expression in a closure.
```suggestion
      Tensor<Scalar>._pullbackSubscript(v, indexPath, shape)
```